File: ChibuzorTasks/mydairy/src/mydairy/DairyEntry.java
package mydairy.src.mydairy;

import java.io.Serial;
import java.time.LocalDate;
import java.io.Serializable;

public class DairyEntry implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;
    private int id;
    private String title;
    private String body;
    private LocalDate date;

    public DairyEntry(int id, String title, String body) {
        this.id = id;
        setTitle(title);
        setBody(body);
        this.date = LocalDate.now();
    }

    public int getId() {
        return id;
    }

    public void setTitle(String title) {
        if(title == null || title.isEmpty()) throw new IllegalArgumentException("Title cannot be null or empty");
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    public void setBody(String body) {
        if(body == null || body.isEmpty()) throw new IllegalArgumentException("Body cannot be null or empty");
        this.body = body;
    }

    public LocalDate getDate() {
        return date;
    }

    public String getBody() {
        return body;
    }

    @Override
    public String toString() {
        return String.format("Date: %s\nTitle: %s\nBody: %s\n ",date, title, body);
    }

}

File: PythonClass/classWorks/Feb-2025/dairy/src/dairy_entry.py
from dairy.src.dairy import *

import datetime

class DairyEntry:
    def __init__(self, entry_id, title, body):
        self.entry_id = entry_id
        self.title = None
        self.body = None
        self.set_title(title)
        self.set_body(body)
        self.date = datetime.date.today()

    def get_entry_id(self):
        return self.entry_id

    def set_title(self, title):
        if not title:
            raise ValueError("Title cannot be null or empty")
        self.title = title

    def get_title(self):
        return self.title

    def set_body(self, body):
        if not body:
            raise ValueError("Body cannot be null or empty")
        self.body = body

    def get_date(self):
        return self.date

    def get_body(self):
        return self.body

    def __str__(self):
        return f"Date: {self.date}\nTitle: {self.title}\nBody: {self.body}\n"
